### DISCUSSION TITLE
Highlighting objects

### DISCUSSION THREAD

**Discussion by CarlSandstrom** [vsg-dev/VulkanSceneGraph#677] [Q&A]
Hi!

I am currently working on a pre-processor for a computational software and currently I am looking on how to use Vsg for displaying 3D objects. When I have selected a component in either the component tree och in the 3D view (not yet implemented), is there a nice way to highlight the selection in the 3D view? I have successfully changed the color of the component by using ..propoerties.dataVariance=DYNAMIC_DATA and change the color data of the node, but then I have to keep track of the old colors. This will of course only change the colors and it would be nice to change the width of the edges of the selected element or something similar to really make it stand out. Is there a good way the highlight an existing object in the scene?

**Comment by robertosfield** [comment]
There isn't a built in way feature that you can enable to add highlighting, instead you'll need to come up with your own scheme for highlighting using the rendering features the the VSG provides.

This is a feature that I expect many users will eventually want to implement so a  vsgExample would be helpful, so if anyone want to pitch in then please go ahead.  I will try to get to this but can't tackle this right away.

How to implement highlighting depends upon how you want to highlight an object or objects in the scene.  Possibilities are:

1. - Render a wireframe box around the object(s) you want to highlight.
2. - Render a wireframe version of the object(s) as well as the original to create a scribe effect.
3. - Render an outline/halo of the object(s)

Option 1 would the most straight forward as it doesn't require any modification of existing state required to render.  Options 2 & 3 would be harder as they were require substitution of state in order to render the subrgaphs.  

@CarlSandstrom did you have a highlighting effect in mind?

**Reply by CarlSandstrom** [reply]
Our current software behaves as shown in the image below and I would like to be able to do something similar. I would say that is an outline effect and change of color of the surface.

![image](https://user-images.githubusercontent.com/3179136/211211649-54729581-d9eb-42a4-a659-4499433da0ce.png)

I guess one way to do it is to make a deep copy of the node parent to that surface, change the color of the surface and the edges and then switch from the original to the copy when it is selected. Maybe multiplying each component of the color in the copy with a value. E.g. [0.9, 0, 0] to make it red. So, in case that works, that leaves two questions; 1) Is there support for making a deep-copy of a node and 2) Is it possible to make an edge bold?

**Comment by WdsPjm** [comment]
hello！
Now is there a solution for Highlighting objects
